### ISSUE TITLE
4.13.0 appears to break dependsOn (possible regression)

### ISSUE THREAD

**Issue by sidawson** [laravel/nova-issues#4751]
- Laravel Version: 8.83.23
- Nova Version: 4.13.0
- PHP Version: 8.1.9
- Database Driver & Version: mariadb 10.8.3
- Operating System and Version: macOS 12.3.1
- Browser type and version: Chrome 103.0.5060.114
- Reproduction Repository: https://github.com/###/###

### Description:

`BelongsTo` with a `Select::make` using `->dependsOn` worked on 4.12.14. Does not work on 4.13.0

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Create a nova page, based on a table that has two interdependent fields.

eg, 
-   table `ticket` - has a `customer_id` and a `customer_contract_id`. 
-  `ticket HasMany customers`
-  `customer HasMany customer_contracts`

```

use App\Models\Customer;
use App\Models\CustomerContract;
use Laravel\Nova\Fields\BelongsTo;
use Laravel\Nova\Fields\FormData;
use Laravel\Nova\Fields\Select;
use Laravel\Nova\Http\Requests\NovaRequest;

BelongsTo::make('Customer'),

// display
BelongsTo::make('Customer Contract', 'customer_contract')
    ->exceptOnForms(),

// edit
Select::make('Customer Contract', 'customer_contract_id')
    ->dependsOn('customer', function(Select $field, NovaRequest $request, FormData $formData) {
        $customer = Customer::whereId($formData->customer ?? $this->customer_id)->first();
        $field->options($customer?->customer_contracts->pluck('description', 'id'));
    })
    ->onlyOnForms(),

```

2. Select a Customer (ie, any element from the parent table dropdown)

#### Expected behaviour (seen only in nova 4.12.14)
When a Customer is selected, the Customer Contract dropdown is limited to contracts owned by that customer

#### Observed behaviour (seen only in nova 4.13.0)
When a Customer is selected, the Customer Contract dropdown disappears from the screen completely

**Comment by crynobone** [comment]
Please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example)

**Comment by sidawson** [comment]
Help me out, I've never reported a nova bug before. So I'd need to create:

1. an empty repo with a clean laravel + nova install (plus all dependencies etc)
2. migrations - including i) 3 x table creations ii) various test data for each of those tables 
3. configuration necessary for said database (so, user roles, passwords etc etc)
4. a single nova page with the above 15 lines of code, plus generic boilerplate

I shove all that in a repo somewhere, then add a link here, THEN the above issue can be looked at? 

Is that correct?

**Comment by crynobone** [comment]
Yes, that's correct. However, you could also make sure you does have the latest aasets installed by running `php artisan nova:publish`.

**Comment by kusab85** [comment]
Hi @crynobone!

I've got the same issue. Here's my config and repo:

- Laravel Version: 9.27.0
- Nova Version: 4.13.00 (Silver Surfer)
- PHP Version: 8.0.11
- Database Driver & Version: SqlLite
- Operating System and Version: Windows 10 21H2
- Browser type and version: MS Edge 104.0.1293.63 64-bit and FireFox 91.13.0esr 64bit
- Reproduction Repository: https://github.com/kusab85/TestApp/tree/nova-dependsOn-belongsTo-field

Open Posts page and edit a Post.
Callback function is called **only once** when page is being loaded.
After that, changing value of the User field does not do a thing.
Just look into laravel.log 

Cheers!

**Comment by crynobone** [comment]
Closing this for now, no one else have experience this issue which made it more possible that this is a usage issue and not a bug.

Feel free to submit new issue with reproducing step if anyone have similar issue.
